paasta_tools/kubernetes_tools.py

```diff
@@ -570,18 +570,25 @@ def namespace_external_metric_name(self, metric_name: str) -> str:
     def get_autoscaling_metric_spec(
         self, name: str, cluster: str, namespace: str = "paasta"
     ) -> Optional[V2beta1HorizontalPodAutoscaler]:
+        # Returns None if an HPA should not be attached based on the config,
+        # or the config is invalid.
+
+        if self.get_desired_state() == "stop":
+            return None
+
+        autoscaling_params = self.get_autoscaling_params()
+        if autoscaling_params["decision_policy"] == "bespoke":
+            return None
+
         # use new autoscaling configuration if it exists.
         if "horizontal_autoscaling" in self.config_dict:
             return self.get_hpa_metric_spec(name, cluster, namespace)
+
         min_replicas = self.get_min_instances()
         max_replicas = self.get_max_instances()
-        if not min_replicas or not max_replicas:
-            log.error(
-                f"Please specify min_instances and max_instances for autoscaling to work: {min_replicas}, {max_replicas}"
```

Comment:
why remove this?  I guess just because no one's ever going to see it?

Reply:
We're calling this method on every update, so it's not really an error if a service just has `instances` instead of min/max.